apps/commerce-api/src/test/kotlin/com/loopers/application/payment/PaymentFacadeConcurrencyTest.kt

```
@@ -0,0 +1,190 @@
+            val latch = CountDownLatch(threadCount)
+
+            val successCount = AtomicInteger(0)
+            val failureCount = AtomicInteger(0)
+
+            // when - 5개 스레드가 동시에 같은 결제를 처리 시도
+            repeat(threadCount) {
+                executor.submit {
+                    try {
+                        paymentFacade.processInProgressPayment(payment.id)
+                        successCount.incrementAndGet()
+                    } catch (e: ObjectOptimisticLockingFailureException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)